### PR TITLE
chore(all): Fix slow dev reload and update NX

### PR DESCRIPTION
**Pull request by snaerseljan**
# Updates NX and fixes slow reloads on our dev servers

## What

- Updates nx from 16.5.0 -> 16.9.0
- Update custom tool executor to support new version in dev mode
- Updates all of our `ts-jest` config to use `isolatedModules: true`, which disables type-checking, which results in much faster test runs. _The main reason for this change is that after the update some tests where experiencing timout's in CI and therefore failed tests. Some of our modules are really big and it takes `ts-jest` a good amount of time to transpile all of those TS files before executing those tests. By skipping the typecheck in `ts-jest` when running the tests we speed up the test workflow by a good amount of time! I dare to say half!_

## Why

Both our node dev servers and spa dev servers where too slow in hot reloading the application. This is because the current version of our nx had some faults. Also reloads where being triggered in server application while the actual change happened within a SPA 

## TODO 

- [x] Fix broken storybook, build and dev are broken

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by obmagnusson — APPROVED** [review]
Codeowner files LGTM

File: package.json
Comment (peturgq) [inline package.json:409]: This should probably not be a carat prefixed version.

**Review by peturgq — APPROVED** [review]
Some notes around the cypress version, otherwise very good.

**Review by birkirkristmunds — COMMENTED** [review]
The application is starting up much faster, but I'm stuck on this error when I try to open an application like, http://localhost:4242/umsoknir/faedingarorlof ,so I can test if the reload is working faster. 

[App] Error	9/26/2023, 10:37:08 AM Fetch failure (ApplicationModule.applicationsApi): request to http://localhost:3333/users/**REMOVE_PII: **REMOVE_PII: **REMOVE_PII: 0101307789******/applications?typeId=ParentalLeave failed, reason: connect ECONNREFUSED ::1:3333

On the main branch im not getting this localhost:3333 error

I have tried following:
- removed .cache dist node_modules 
- delete all from Docker
- Run yarn dev-init application-system-form
- Run yarn install and yarn schemas


[localhost3333-log.pdf](https://github.com/island-is/island.is/files/12726494/localhost3333-log.pdf)

**Comment by snaerseljan** [comment]
> The application is starting up much faster, but I'm stuck on this error when I try to open an application like, http://localhost:4242/umsoknir/faedingarorlof ,so I can test if the reload is working faster.
> 
> [App] Error 9/26/2023, 10:37:08 AM Fetch failure (ApplicationModule.applicationsApi): request to http://localhost:3333/users/**REMOVE_PII: **REMOVE_PII: REMOVE_PII: 0101307789****/applications?typeId=ParentalLeave failed, reason: connect ECONNREFUSED ::1:3333
> 
> On the main branch im not getting this localhost:3333 error
> 
> I have tried following:
> 
> * removed .cache dist node_modules
> * delete all from Docker
> * Run yarn dev-init application-system-form
> * Run yarn install and yarn schemas
> 
> [localhost3333-log.pdf](https://github.com/island-is/island.is/files/12726494/localhost3333-log.pdf)

The error indicates that you are not running the service

**Comment by birkirkristmunds** [comment]
> > The application is starting up much faster, but I'm stuck on this error when I try to open an application like, http://localhost:4242/umsoknir/faedingarorlof ,so I can test if the reload is working faster.
> > [App] Error 9/26/2023, 10:37:08 AM Fetch failure (ApplicationModule.applicationsApi): request to http://localhost:3333/users/**REMOVE_PII: **REMOVE_PII: REMOVE_PII: 0101307789****/applications?typeId=ParentalLeave failed, reason: connect ECONNREFUSED ::1:3333
> > On the main branch im not getting this localhost:3333 error
> > I have tried following:
> > 
> > * removed .cache dist node_modules
> > * delete all from Docker
> > * Run yarn dev-init application-system-form
> > * Run yarn install and yarn schemas
> > 
> > [localhost3333-log.pdf](https://github.com/island-is/island.is/files/12726494/localhost3333-log.pdf)
> 
> The error indicates that you are not running the service

I started this just as the main branch, using _yarn dev application-system-form_ , there I don't have this issue.
Is there some other command that I need to use to start the application?

**Comment by saevarma** [comment]
> > > The application is starting up much faster, but I'm stuck on this error when I try to open an application like, http://localhost:4242/umsoknir/faedingarorlof ,so I can test if the reload is working faster.
> > > [App] Error 9/26/2023, 10:37:08 AM Fetch failure (ApplicationModule.applicationsApi): request to http://localhost:3333/users/**REMOVE_PII: **REMOVE_PII: REMOVE_PII: 0101307789****/applications?typeId=ParentalLeave failed, reason: connect ECONNREFUSED ::1:3333
> > > On the main branch im not getting this localhost:3333 error
> > > I have tried following:
> > > 
> > > * removed .cache dist node_modules
> > > * delete all from Docker
> > > * Run yarn dev-init application-system-form
> > > * Run yarn install and yarn schemas
> > > 
> > > [localhost3333-log.pdf](https://github.com/island-is/island.is/files/12726494/localhost3333-log.pdf)
> > 
> > 
> > The error indicates that you are not running the service
> 
> I started this just as the main branch, using _yarn dev application-system-form_ , there I don't have this issue. Is there some other command that I need to use to start the application?

Thanks for reporting this 🙏 There is a startup error in the application-system-api which we are resolving, then testing startup of all apps and services via feature deployment.

**Comment by saevarma** [comment]
Admin merging this PR as CI is green and this requires review from all codeowners which we are not waiting for.

**Pull request merged**
